core/src/main/java/com/linecorp/armeria/common/Cookie.java

```diff
@@ -473,6 +473,18 @@ static Cookies fromSetCookieHeaders(boolean strict, Collection<String> setCookie
     @Nullable
     String sameSite();
 
+    /**
+     * Returns whether this {@link Cookie} should only match its original host in domain matching. If this
+     * returns {@code true}, should {@link #domain()} returns a non-null value, it's going to be the original
+     * request host.
+     */
+    boolean isHostOnly();
+
+    /**
+     * Returns whether this {@link Cookie} has been expired.
+     */
+    boolean isExpired();
```

Comment:
Should we also add the version that accepts currentTimeMillis? i.e. `boolean isExpired(long currentTimeMillis)`

Reply:
Not sure when it's used 🤔 Because we probably don't check expiration until we actually use the cookie.

Reply:
A user may not want `System.currentTimeMillis()` to be invoked multiple times. e.g.

```java
long currentTimeMillis = System.currentTimeMillis();
for (Cookie c : cookies) {
    if (c.isExpired(currentTimeMillis)) {
        System.err.println("Expired: " + c)
    }
}
```